src/category_theory/sites/pretopology.lean

```diff
@@ -70,11 +70,20 @@ namespace pretopology
 
 instance : has_coe_to_fun (pretopology C) (λ _, Π X : C, set (presieve X)) := ⟨coverings⟩
 
+variable {C}
+
+instance : has_le (pretopology C) :=
```

Comment:
What's the reason behind defining those intermediate instances? I can understand when they hold under weakerconditions, but that's not the case here?

Reply:
The reason is that I think some change of mine changed the way that elaboration of `le` defined on `pi` types works. Which makes me have to define a `le_def` iff lemma, so that I can hint more to the elaborator. For an example, check the diff here:
https://github.com/leanprover-community/mathlib/pull/9891/files/b15160537f55fd6cc7bb7c443c8e2aa3390b1975..fc88f2925e691f2de30d36add082dece17f02a04#diff-bf7371cfd6bc489df26b33da7a7b450a755fba2382a871d639c50666c9c0563eR82-R86